src/ZODB/tests/StorageTestBase.py

```diff
@@ -113,18 +113,24 @@ def handle_all_serials(oid, *args):
 
     The original interface just returned the serialno for the
     object.
+
+    The updated multi-commit API returns nothing from store(), and
+    returns a sequence of resolved oids from tpc_vote.
     """
     d = {}
     for arg in args:
         if isinstance(arg, bytes):
             d[oid] = arg
-        elif arg is None:
-            pass
-        else:
-            for oid, serial in arg:
-                if not isinstance(serial, bytes):
-                    raise serial # error from ZEO server
-                d[oid] = serial
+        elif arg:
+            for t in arg:
+                if isinstance(t, bytes):
+                    # This will be the tid returned by tpc_finish.
```

Comment:
> On Jul 12, 2016, at 11:11, Jim Fulton <notifications@github.com> wrote:
> 
> > +                    # This will be the tid returned by tpc_finish.
> 
> It could also be an oid that's returned by vote to indicate that a conflict was resolved. This is important in some situations, but perhaps not in cases that are tested here.
> 
> This is a mess (like a lot of ZODB test infrastructure). For this function to be "correct", it needs to consider where the data came from.

The comment is unclear. I meant that the caller will wind up using the tid returned by tpc_finish in this case (because the dict will be empty). I can clarify.